### PR TITLE
[CL-285] Add default loading state to popup-page

### PR DESCRIPTION
**Pull request by vleague2**
## 🎟️ Tracking

[CL-285](https://bitwarden.atlassian.net/browse/CL-285)

## 📔 Objective

This PR adds an accessible loading state to the `popup-page`. It also reduces some boilerplate code in the stories file.

## 📸 Screenshots

Click on the "Storybook Publish" action and view the Browser > Popup Layout "Loading" story.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[CL-285]: https://bitwarden.atlassian.net/browse/CL-285?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/browser/src/platform/popup/layout/popup-page.component.html [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
+    [ngClass]="{ 'tw-invisible': loading }"
+  >
     <ng-content></ng-content>
   </div>
+  <span
+    class="tw-absolute tw-inset-0 tw-flex tw-items-center tw-justify-center tw-text-main"
+    [ngClass]="{ 'tw-invisible': !loading }"
+  >
+    <i class="bwi bwi-spinner bwi-lg bwi-spin" [ariaLabel]="loadingText"></i>
+  </span>
```
Comment (willmartian) [inline apps/browser/src/platform/popup/layout/popup-page.component.html:19]: What is the expected screen reader functionality when interacting with this component?

File: apps/browser/src/platform/popup/layout/popup-page.component.ts [hunk 2]
```diff
@@ -7,5 +10,15 @@ import { Component } from "@angular/core";
   host: {
     class: "tw-h-full tw-flex tw-flex-col tw-flex-1 tw-overflow-y-auto",
   },
+  imports: [CommonModule],
 })
-export class PopupPageComponent {}
+export class PopupPageComponent implements OnInit {
+  constructor(protected i18nService: I18nService) {}
```
Comment (willmartian) [inline apps/browser/src/platform/popup/layout/popup-page.component.ts:16]: We should default to using the newer `inject` function vs constructor injection

File: apps/browser/src/platform/popup/layout/popup-page.component.ts [hunk 3]
```diff
@@ -7,5 +10,15 @@ import { Component } from "@angular/core";
...
+
+  @Input() loading = false;
+  @Input() loadingText?: string;
+
+  ngOnInit() {
+    this.loadingText = this.loadingText ?? this.i18nService.t("loading");
```
Comment (willmartian) [inline apps/browser/src/platform/popup/layout/popup-page.component.ts:22]: I believe we can just make this the default value when declaring the property.

File: apps/browser/src/platform/popup/layout/popup-page.component.html [hunk 4]
```diff
@@ -1,7 +1,21 @@
 <ng-content select="[slot=header]"></ng-content>
-<main class="tw-bg-background-alt tw-flex-1 tw-overflow-y-auto tw-h-full">
-  <div class="tw-max-w-screen-sm tw-mx-auto tw-p-3 tw-overflow-y-auto tw-h-full">
+<main
+  class="tw-flex-1 tw-overflow-y-auto tw-h-full tw-relative"
+  [ngClass]="loading ? 'tw-bg-black/30' : 'tw-bg-background-alt'"
```
Comment (willmartian) [inline apps/browser/src/platform/popup/layout/popup-page.component.html:4]: There was a miscommunication regarding the desired end behavior here. We can just use `tw-bg-background-alt`

**Review by willmartian — CHANGES_REQUESTED** [review]

File: apps/browser/src/platform/popup/layout/popup-page.component.html (hunk 1 above)
Comment (vleague2) [reply apps/browser/src/platform/popup/layout/popup-page.component.html]: After much discussion, we have decided to go with a simple aria-label on the spinner for now, and defer further accessibility implementation to another ticket.

File: apps/browser/src/platform/popup/layout/popup-page.component.ts [hunk 5]
```diff
@@ -7,5 +10,13 @@ import { Component } from "@angular/core";
...
+  imports: [CommonModule],
 })
-export class PopupPageComponent {}
+export class PopupPageComponent {
+  protected i18nService = inject(I18nService);
+
+  @Input() loading = false;
+
+  /** Accessible loading label for the spinner. Defaults to "loading" */
+  @Input() loadingText?: string = this.i18nService.t("loading");
```
Comment (willmartian) [inline apps/browser/src/platform/popup/layout/popup-page.component.ts:21]: Do you think adding the ability to customize this now is still necessary if we are revisiting in the future?

Comment (vleague2) [reply apps/browser/src/platform/popup/layout/popup-page.component.ts:21]: Maybe not necessary, but if we don't end up returning to this accessibility work in the near future, I think it'd be nice to have the ability to customize it for design in the meantime.

File: apps/browser/src/_locales/en/messages.json [hunk 6]
```diff
@@ -3585,5 +3585,8 @@
         "example": "Visa"
       }
     }
+  },
+  "loading": {
+    "message": "Loading"
```
Comment (willmartian) [inline apps/browser/src/_locales/en/messages.json:3590]: Surprised we didn't already have this...

**Review by willmartian — APPROVED** [review]
LGTM :shipit:

File: apps/browser/src/_locales/en/messages.json (hunk 6 above)
Comment (vleague2) [reply apps/browser/src/_locales/en/messages.json:3590]: I saw it in the web app but not in browser :(

**Review by danielleflinn — APPROVED** [review]

**Review by willmartian — APPROVED** [review]

**Pull request merged**